lib/redix/stream/consumer.ex

```diff
@@ -22,55 +29,143 @@ defmodule Redix.Stream.Consumer do
   Initializes a new `Redix.Stream.Consumer`, establishing a long-term
   stream with the given `redis` server.
   """
-  @spec init({Redix.Stream.redix, Redix.Stream.t, function() | mfa(), keyword}) :: {:ok, state}
+  @spec init({Redix.Stream.redix(), Redix.Stream.t(), function() | mfa(), keyword}) ::
+          {:ok, state}
   def init({redix, stream, handler, opts}) do
     timeout = Keyword.get(opts, :timeout, @default_timeout)
-    default_start_pos = Keyword.get(opts, :start_pos, "$")
-    tracker = Keyword.get(opts, :tracker, nil)
-
-    # If tracker specified, load our starting position
-    start_pos = if tracker do
-      case Redix.command!(redix, ["GET", "stream_tracker:#{tracker}"]) do
-        nil -> default_start_pos
-        pos -> pos
+    group_name = Keyword.get(opts, :group_name)
+    consumer_name = Keyword.get(opts, :consumer_name)
+    consumer_group_command_connection = Keyword.get(opts, :consumer_group_command_connection)
+
+    default_start_pos =
+      case group_name do
+        nil -> "$"
+        _ -> ">"
+      end
+
+    start_pos = Keyword.get(opts, :start_pos, default_start_pos)
+
+    if consumer_name do
+      try do
+        start_pos =
+          case start_pos do
+            "$" -> "$"
+            ">" -> "$"
+            other -> other
+          end
+
+        Redix.command(redix, ["XGROUP", "CREATE", stream, group_name, start_pos])
```

Comment:
I wonder if we want to keep init pure, since we might want to start this process before we've ensured our connection to redis is up and running.